### ISSUE TITLE
Null case of @Default not covered

### ISSUE THREAD

**Issue by southernkasaist** [line/armeria#1858]
https://github.com/line/armeria/blob/61a72e99c9a53008effc1e55ac63567fefc4ebce/core/src/main/java/com/linecorp/armeria/internal/annotation/AnnotatedValueResolver.java#L1026

When intent to use `null` for default value like `@Param("foo") @Default FooType foo`, `AnnotatedValueResolver` doesn't check if `Optional.isPresent` but directly get underlying value, which will be ended up with `No value present`.

Expected is following:
```java
final Optional<String> specifiedValue = getSpecifiedValue(aDefault.value());
defaultValue = specifiedValue.isPresent() ? specifiedValue.get() : null;
```

**Comment by trustin** [comment]
Doh! Are you interested in sending a pull request, or shall I send one based on your suggestion?

**Comment by southernkasaist** [comment]
Seems that to write correct test cases, I need to take time to have comprehensive understanding of test structures.

So I appreciate if you can take it instead 🙇

**Comment by trustin** [comment]
Sure. No problem!

**Comment by trustin** [comment]
Will be fixed via #1864
